tests/phpUnit/Catrobat/Controller/Admin/LogLineTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Tests\phpUnit\Catrobat\Controller\Admin;
+
+use App\Catrobat\Controller\Admin\LogLine;
+use App\Catrobat\Controller\Admin\LogsController;
+use PHPUnit\Framework\MockObject\MockObject;
+use ReflectionException;
+use Tests\phpUnit\CatrowebPhpUnit\CatrowebTestCase;
+
+/**
+ * @internal
+ * @covers \App\Catrobat\Controller\Admin\LogLine
+ */
+class LogLineTest extends CatrowebTestCase
+{
+  private MockObject $object;
+
+  protected function setUp(): void
+  {
+    $this->object = $this->getMockForAbstractClass(LogLine::class);
```

Comment:
Why abstract?

Reply:
was easier to create one mock object for all class methods (public and private) and then invoke them in every single test function. otherwise i'd create one real object of the class for public methods for example:
$this->object = $this->createMock(LogLine::class); --> that would be for private and protected methods
and this $logLine = new LogLine(); for public methods.
so what do you think ?

Reply:
Thx for the explanation. Sounds fine to me!